packages/hdwallet-gridplus/src/bitcoin.ts

```diff
@@ -0,0 +1,711 @@
+import { Client, Constants } from "gridplus-sdk";
+
+import { UTXO_NETWORK_PARAMS } from "./constants";
+import { deriveAddressFromPubkey } from "./utils";
+
+const u32le = (n: number): Buffer => {
+  const b = Buffer.alloc(4);
+  b.writeUInt32LE(n >>> 0, 0);
+  return b;
+};
+
+const scriptTypeToPurpose = (scriptType: core.BTCInputScriptType): number => {
+  switch (scriptType) {
+    case core.BTCInputScriptType.SpendAddress:
+      return 44;
+    case core.BTCInputScriptType.SpendP2SHWitness:
+      return 49;
+    case core.BTCInputScriptType.SpendWitness:
+      return 84;
+    default:
+      return 44;
+  }
+};
+
+const encodeDerInteger = (x: Buffer): Buffer => {
+  if (x[0] & 0x80) {
+    return Buffer.concat([Buffer.from([0x00]), x]);
+  }
+  return x;
+};
+
+const encodeDerSignature = (rBuf: Buffer, sBuf: Buffer, sigHashType: number): Buffer => {
+  const rEncoded = encodeDerInteger(rBuf);
+  const sEncoded = encodeDerInteger(sBuf);
+
+  const derSignature = Buffer.concat([
+    Buffer.from([0x30]),
+    Buffer.from([rEncoded.length + sEncoded.length + 4]),
+    Buffer.from([0x02]),
+    Buffer.from([rEncoded.length]),
+    rEncoded,
+    Buffer.from([0x02]),
+    Buffer.from([sEncoded.length]),
+    sEncoded,
+    Buffer.from([sigHashType]),
+  ]);
+
+  return derSignature;
+};
+
+export const btcGetAccountPaths = (msg: core.BTCGetAccountPaths): Array<core.BTCAccountPath> => {
+  const slip44 = core.slip44ByCoin(msg.coin);
+  if (!slip44) throw new Error(`Unsupported coin: ${msg.coin}`);
+
+  const scriptTypes: core.BTCInputScriptType[] = (() => {
+    if (msg.coin === "Dogecoin" || msg.coin === "BitcoinCash") {
+      return [core.BTCInputScriptType.SpendAddress];
+    } else {
+      return [
+        core.BTCInputScriptType.SpendAddress,
+        core.BTCInputScriptType.SpendP2SHWitness,
+        core.BTCInputScriptType.SpendWitness,
+      ];
+    }
+  })();
+
+  return scriptTypes.map((scriptType) => {
+    const purpose = scriptTypeToPurpose(scriptType);
+    return {
+      coin: msg.coin,
+      scriptType,
+      addressNList: [0x80000000 + purpose, 0x80000000 + slip44, 0x80000000 + (msg.accountIdx || 0), 0, 0],
+    };
+  });
+};
+
+export async function btcGetAddress(client: Client, msg: core.BTCGetAddress): Promise<string | null> {
+  // Get compressed public key from device (works for all UTXOs)
+  // Using SECP256K1_PUB flag bypasses Lattice's address formatting,
+  // which only supports Bitcoin/EVM chains/Solana
+  const pubkeys = await client.getAddresses({
+    startPath: msg.addressNList,
+    n: 1,
+    flag: Constants.GET_ADDR_FLAGS.SECP256K1_PUB,
+  });
+
+  if (!pubkeys || !pubkeys.length) {
+    throw new Error("No public key returned from device");
+  }
+
+  // pubkeys[0] may be uncompressed (65 bytes) or compressed (33 bytes)
+  const pubkeyBuffer = Buffer.isBuffer(pubkeys[0]) ? pubkeys[0] : Buffer.from(pubkeys[0], "hex");
+
+  // Compress if needed (65 bytes = uncompressed, 33 bytes = already compressed)
```

Comment:
was this validated as possible for any utxo coins? looking at sourcecode, it appears utxo chains would always return the secp256k1 pubkey bytes.